buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -706,6 +706,66 @@ public static void copy(AsciiString src, int srcIdx, ByteBuf dst, int length) {
     }
 
     /**
+     * Creates a retained duplicate of the specified {@link ByteBuf}.
+     */
+    public static ByteBuf rduplicate(ByteBuf buf) {
+        final int readerIndex = buf.readerIndex();
+        final int writerIndex = buf.writerIndex();
+
+        final AbstractByteBuf duplicate;
+        if (buf instanceof SlicedAbstractByteBuf) {
+            final SlicedAbstractByteBuf oldSlice = (SlicedAbstractByteBuf) buf;
+            duplicate = PooledSlicedByteBuf.newInstance(oldSlice.unwrap(), 0, buf.capacity(), oldSlice.adjustment());
```

Comment:
Any suggestions WRT what the sensible behavior should be? We could create pooled slices only when the current allocator is pooled, but perhaps we could just leave this as is given that this is a new functionality.